src/Microsoft.AspNetCore.ResponseCompression/ResponseCompressionMiddleware.cs

```diff
@@ -53,14 +53,13 @@ public ResponseCompressionMiddleware(RequestDelegate next, IResponseCompressionP
         /// <returns></returns>
         public async Task Invoke(HttpContext context)
         {
-            ICompressionProvider compressionProvider = null;
-
+            var shouldCompress = false;
             if (!context.Request.IsHttps || _enableForHttps)
```

Comment:
Move the https check inside ShouldCompressRequest so devs can override it per request.